File: sdks/go/src/core/sink.go
package core

import "context"

// Sink receives encoded events and delivers them to a destination.
// All methods must be safe for concurrent use.
type Sink interface {
	// Name returns a human-readable identifier for this sink.
	Name() string
	// WriteEvent delivers an already-encoded event to the sink.
	// encoded is the JSON bytes (including trailing newline).
	// ev is the original Event for sinks that need typed field access.
	WriteEvent(ctx context.Context, encoded []byte, ev *Event) error
	// Flush forces any buffered data to be written.
	Flush(ctx context.Context) error
	// Close releases resources held by the sink.
	Close(ctx context.Context) error
}

File: sdks/js/src/sinks/sink.ts
/** Sink interface — destination for encoded events. */
export interface Sink {
  name(): string;
  write(encoded: string): Promise<void> | void;
  flush(): Promise<void> | void;
  close(): Promise<void> | void;
  drain?(): Promise<void> | void;
  pause?(): void;
  resume?(): void;
  queueSize?(): number;
  health?(): boolean | Promise<boolean>;
}
